### PR TITLE
Update eval.js

### PR DESCRIPTION
**Pull request by Stitch07**
**Please describe the changes this PR makes and why it should be merged, as well as any and all proof of successful tests:**

This makes an important command usable again.

**Semantic versioning classification:**  
- [ ] This PR changes the framework's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [X] This PR fixes messes caused by Soumil07 directly committing.
- [ ] This PR **only** includes non-code changes, like changes to README, etc.

### CODE REVIEW

File: commands/eval.js [hunk 1]
```diff
@@ -6,36 +6,33 @@
...
-    } catch (err) {
-      message.channel.send(`\`ERROR\` \`\`\`xl\n${await this.client.clean(this.client, err)}\n\`\`\``);
+exports.run = async (client, message, args, level) => { // eslint-disable-line no-unused-vars
+  const code = args.join(" ");
+  try {
+    const evaled = eval(code);
+    const clean = await client.clean(client, evaled);
+    // sends evaled output as a file if it exceeds the maximum character limit
+    // 6 graves, and 2 characters for "js"
+    const MAX_CHARS = 3 + 2 + clean.length + 3;
```
Comment (kyranet) [inline commands/eval.js]: Except when the output has 1991-1992 characters, where 3 + 2 + 1991 + 3 is 1999, plus the two new lines inside the codeblock, 2001. Please consider wrapping the codeblock in clean for maximum accuracy and make it less confusing.

File: commands/eval.js [hunk 2]
```diff
@@ -6,36 +6,33 @@
...
+    if (MAX_CHARS > 2000) {
+      message.channel.send("Output exceeded 2000 characters. Sending as a file.", { files: [{ attachment: Buffer.from(clean), name: "output.txt" }] });
     }
+    message.channel.send(`\`\`\`js\n${clean}\n\`\`\``);
+  } catch (err) {
+    message.channel.send(`\`ERROR\` \`\`\`xl\n${await client.clean(client, err)}\n\`\`\``);
```
Comment (kyranet) [inline commands/eval.js]: The codeblock should use js syntax, and this does not watch if the length is > 2000 chars, as well as it has code duplication (client.clean is used twice).

**Review by kyranet — CHANGES_REQUESTED** [review]
Please consider wrapping the eval itself inside a try/catch for better handling and less code duplication.

File: commands/eval.js [hunk 3]
```diff
@@ -3,18 +3,37 @@
...
-
 // However it's, like, super ultra useful for troubleshooting and doing stuff
 // you don't want to put in a command.
 exports.run = async (client, message, args, level) => { // eslint-disable-line no-unused-vars
-  const code = args.join(" ");
+  const { output, error } = this.tryEval(args.join(" "));
+  const cleaned = await client.clean(client, output);
+  // sends output as a file if it exceeds the character limit
+  // note: if the output is an error, it will have the ERROR heading, hence the extra 7 
+  const MAX_CHARS = error ? 3 + 2 + 7 + cleaned.length + 2 + 3 : 3 + 2 + cleaned.length + 2 + 3;
```
Comment (MrJacz) [inline commands/eval.js]: this can be `const MAX_CHARS = error ? 17 + cleaned.length : cleaned.length + 10;`

**Review by MrJacz — APPROVED** [review]

File: commands/eval.js (hunk 3 above)
Comment (Stitch07) [reply commands/eval.js]: I did that so it would server as an explanation to people viewing the code, but okay.

File: commands/eval.js [hunk 4]
```diff
@@ -3,18 +3,37 @@
 // can be used to do **anything** on your machine, from stealing information to
 // purging the hard drive. DO NOT LET ANYONE ELSE USE THIS
 
-
 // However it's, like, super ultra useful for troubleshooting and doing stuff
 // you don't want to put in a command.
 exports.run = async (client, message, args, level) => { // eslint-disable-line no-unused-vars
-  const code = args.join(" ");
+  const { output, error } = this.tryEval(args.join(" "));
+  const cleaned = await client.clean(client, output);
```
Comment (kyranet) [inline commands/eval.js:10]: Add the codeblock + output/error message to cleaned, so you have the actual length, and you won't need any weird maths to check the message length.

File: commands/eval.js [hunk 5]
```diff
@@ -3,18 +3,37 @@
...
+  // sends output as a file if it exceeds the character limit
+  // note: if the output is an error, it will have the ERROR heading, hence the extra 7 
+  const MAX_CHARS = error ? 17 + cleaned.length : 10 + cleaned.length;
+  if (MAX_CHARS > 2000) {
+    message.channel.send("Output exceeded 2000 characters. Sending as a file.", { files: [{ attachment: Buffer.from(cleaned), name: "output.txt" }] });
```
Comment (kyranet) [inline commands/eval.js:15]: Should return. At this moment, it'll send the file and another message (failing due to max length).

File: commands/eval.js [hunk 6]
```diff
@@ -3,18 +3,37 @@
...
+// if it errors, it will return the output as the error and "error" as true
+exports.tryEval = code => {
+  let output, error;
   try {
-    const evaled = eval(code);
-    const clean = await client.clean(client, evaled);
-    message.channel.send(`\`\`\`js\n${clean}\n\`\`\``);
-  } catch (err) {
-    message.channel.send(`\`ERROR\` \`\`\`xl\n${await client.clean(client, err)}\n\`\`\``);
+    output = eval(code);
```
Comment (kyranet) [inline commands/eval.js:26]: If it's a promise, that'll resolve, but later, if it throws, it'll still show that it didn't error.

**Review by kyranet — CHANGES_REQUESTED** [review]
Last requested changes.

**Pull request closed without merging**